c/src/measure/stats/systemstats.cpp

```diff
@@ -413,17 +421,32 @@ Stats SystemStats::getInfo() {
 	cpuInfo.modelname = getSysctl<std::string>("machdep.cpu.brand_string");
 #endif
 
-	std::string caches = "";
+	std::stringstream cachesStream;
+	cachesStream << "{";
 	size_t cacheIdx = 1;
+	bool first = true;
 	for (auto& [unified, instruct, data] : cpuInfo.caches) {
-		if (unified)
-			caches += _fmt::format("\"l{}\": \"{} KiB\",", cacheIdx, unified / 1024);
-		if (instruct)
-			caches += _fmt::format("\"l{}i\": \"{} KiB\",", cacheIdx, instruct / 1024);
-		if (data)
-			caches += _fmt::format("\"l{}d\": \"{} KiB\",", cacheIdx, data / 1024);
+		if (unified) {
+			if (!first)
+				cachesStream << ",";
+			cachesStream << _fmt::format("\"l{}\": {}", cacheIdx, unified);
+			first = false;
+		}
+		if (instruct) {
+			if (!first)
+				cachesStream << ",";
+			cachesStream << _fmt::format("\"l{}i\": {}", cacheIdx, instruct);
+			first = false;
+		}
+		if (data) {
+			if (!first)
+				cachesStream << ",";
+			cachesStream << _fmt::format("\"l{}d\": {}", cacheIdx, data);
+			first = false;
+		}
```

Comment:
To remove all the conditions, I would preferr a `delim_seq` that is initially `""` and the set to `","`. Maybe something like this:

```suggestion
		if (unified) {
			cachesStream << delim << _fmt::format("\"l{}\": {}", cacheIdx, unified);
			delim = ",";
		}
		if (instruct) {
			cachesStream << delim << _fmt::format("\"l{}i\": {}", cacheIdx, instruct);
			delim = ","
		}
		if (data) {
			cachesStream << delim << _fmt::format("\"l{}d\": {}", cacheIdx, data);
			delim = ",";
		}
```

`_fmt::format` may also not be necessary anymore since we use stringstream here

Reply:
Important consideration: Note that `ir_metadata` always gives *string values* where the unit is appended seoarated by a space (see for example for RAM [here](https://www.ir-metadata.org/metadata/platform/)). Our previous extension of `ir_metadata` respected this design decision and adhered to it. It would be removed with the changes proposed here.